### PR TITLE
Closes #233 324 332 admiraldev miscellanous documentation updates

### LINKED ISSUE #233: Documentation: Describe how quoting is used in admiral
### Please select a category the issue is focused on?

Developer Guides

### Let us know where something needs a refresh or put your idea here!

In the programming strategy it should be described how quoting is used in admiral, e.g.,

- quoting of arguments which expect unquoted expressions like `new_var` or `filter`.
- how to use quoted values in function calls or when constructing other expressions. (`!!`, `!!!`, `expr_c`, ...)

### LINKED ISSUE #324: Documentation: Add Usage of Footnotes to "Writing Vignettes"
### Please select a category the issue is focused on?

Developer Guides

### Let us know where something needs a refresh or put your idea here!

A section regarding footnotes should be added to the "Writing Vignettes" vignette. The section should provide guidance when footnotes should be used (for information which is useful but does not fit in the main flow of the text, e.g., the source of the test data or which patients were selected for the examples) and a small example.

### LINKED ISSUE #332: Documentation:  Explain standardization of roxygen text using functions
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

See https://github.com/pharmaverse/admiral/pull/2065

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #233, Closes #324, Closes #332 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
We should have this go to `main`.  Going to purge `devel` in the great GitHub Flow update of 2023

**Comment by bms63** [comment]
Is this ready for a review?

**Comment by zdz2101** [comment]
> Is this ready for a review?
Yes

File: vignettes/programming_strategy.Rmd [hunk 1]
```diff
@@ -739,6 +739,17 @@ Please take the following list as recommendation and try to adhere to its rules
...
 
+## How Quoting is used
+* Some admiral arguments require selecting one particular option like `mode`, e.g. `mode = "last"`. Use quotation marks to capture these. The expected assertion function corresponding to these arguments is `assert_character_scalar()/assert_character_vector()`.
+* Many admiral arguments require capturing an expression which are to be evaluated _later_ inside the function body, see arguments like `new_vars`, e.g. `new_vars = exprs(TRTSDTM = EXSTDTM)`. Oftentimes, the assertion function corresponding to these are `assert_expr()/assert_expr_list()`
+* Use `!!` or `!!!` to unquote the expression(s) as needed. Keep in mind `!!` is a one-to-one replacement and `!!!` is a one-to-many replacement. Please see [this chapter](https://adv-r.hadley.nz/quasiquotation.html) in the Advanced R textbook for more details.
+
+## Standardizing Text Used to Label and Describe Arguments
+
+In the following [PR](https://github.com/pharmaverse/admiral/pull/2065), you will find an example of how the function argument `dataset` was able to be standardized such that the Label and Description of said function argument was aligned across the codebase.
```
Comment (manciniedoardo) [inline vignettes/programming_strategy.Rmd:749]: Minor, but could you please add something specifying where to look within this PR? Along the lines of "Please see the changes to `derive_adeg_params()`".

File: vignettes/programming_strategy.Rmd [hunk 2]
```diff
@@ -739,6 +739,17 @@ Please take the following list as recommendation and try to adhere to its rules
 * `dplyr::if_else()` should be used when there are only two conditions. 
 Try to always set the `missing` argument whenever appropriate.
 
+## How Quoting is used
```
Comment (manciniedoardo) [inline vignettes/programming_strategy.Rmd:742]: This is a great explanatory paragraph, @zdz2101 - I am wondering if we can link to it somewhere in admiral? It would be useful for users starting to learn our framework.

File: vignettes/programming_strategy.Rmd [hunk 3]
```diff
@@ -739,6 +739,17 @@ Please take the following list as recommendation and try to adhere to its rules
...
+* Many admiral arguments require capturing an expression which are to be evaluated _later_ inside the function body, see arguments like `new_vars`, e.g. `new_vars = exprs(TRTSDTM = EXSTDTM)`. Oftentimes, the assertion function corresponding to these are `assert_expr()/assert_expr_list()`
+* Use `!!` or `!!!` to unquote the expression(s) as needed. Keep in mind `!!` is a one-to-one replacement and `!!!` is a one-to-many replacement. Please see [this chapter](https://adv-r.hadley.nz/quasiquotation.html) in the Advanced R textbook for more details.
+
+## Standardizing Text Used to Label and Describe Arguments
+
+In the following [PR](https://github.com/pharmaverse/admiral/pull/2065), you will find an example of how the function argument `dataset` was able to be standardized such that the Label and Description of said function argument was aligned across the codebase. 
+
+The benefits of having a programmatic way to write documentation is that if any changes need to be made, making the modification on the corresponding function, in this case, `roxygen_param_dataset()`, scales across the codebase, can be tested, and is less prone to user-error such as typos or grammar mistakes.
```
Comment (manciniedoardo) [inline vignettes/programming_strategy.Rmd:751]: Another minor comment: I think this reads better with brackets: _[...] making the modification on the corresponding function (in this case, `roxygen_param_dataset()`) scales across the codebase, can be tested, [...]_

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks for working on this @zdz2101!

File: vignettes/programming_strategy.Rmd [hunk 4]
```diff
@@ -739,6 +739,17 @@ Please take the following list as recommendation and try to adhere to its rules
 * `dplyr::if_else()` should be used when there are only two conditions. 
 Try to always set the `missing` argument whenever appropriate.
 
+## How Quoting is used
+* Some admiral arguments require selecting one particular option like `mode`, e.g. `mode = "last"`. Use quotation marks to capture these. The expected assertion function corresponding to these arguments is `assert_character_scalar()/assert_character_vector()`.
+* Many admiral arguments require capturing an expression which are to be evaluated _later_ inside the function body, see arguments like `new_vars`, e.g. `new_vars = exprs(TRTSDTM = EXSTDTM)`. Oftentimes, the assertion function corresponding to these are `assert_expr()/assert_expr_list()`
```
Comment (bundfussr) [inline vignettes/programming_strategy.Rmd:744]: I would distinguish two cases:
- Arguments like `new_var` or `filter` which expect a single variable or expression. Here the value is not quoted in the call. In the function it has to be quoted by using `enexpr()`. Usually this is combined with the assertion, e.g., `new_var <- assert_symbol(enexpr(new_var))`. These arguments are unquoted by using `!!`.
- Arguments like `new_vars` or `by_vars` which expect more than one variable or expression. Here the value is quoted in the call (by using `exprs()`). These arguments are unquoted by using `!!!`.

File: vignettes/programming_strategy.Rmd [hunk 5]
```diff
@@ -739,6 +739,17 @@ Please take the following list as recommendation and try to adhere to its rules
...
+* Use `!!` or `!!!` to unquote the expression(s) as needed. Keep in mind `!!` is a one-to-one replacement and `!!!` is a one-to-many replacement. Please see [this chapter](https://adv-r.hadley.nz/quasiquotation.html) in the Advanced R textbook for more details.
+
+## Standardizing Text Used to Label and Describe Arguments
+
+In the following [PR](https://github.com/pharmaverse/admiral/pull/2065), you will find an example of how the function argument `dataset` was able to be standardized such that the Label and Description of said function argument was aligned across the codebase. 
+
+The benefits of having a programmatic way to write documentation is that if any changes need to be made, making the modification on the corresponding function, in this case, `roxygen_param_dataset()`, scales across the codebase, can be tested, and is less prone to user-error such as typos or grammar mistakes.
+
```
Comment (bundfussr) [inline vignettes/programming_strategy.Rmd:752]: I would add that these functions are implemented in `roxygen2.R`.

Should we add a naming convention for the functions?

File: vignettes/writing_vignettes.Rmd [hunk 6]
```diff
@@ -181,6 +179,25 @@ restricted either by restricting the source datasets at the beginning of the
 vignette or in the call of `dataset_vignette()` if only some calls are
 affected.
 
+#### Using Footnotes
+
+Footnotes can be useful to add context without adding clutter to the primary subject matter of the vignette being written.
+
+To use footnotes, add a caret and an identifier inside brackets e.g. ([^1]). The identifiers can be numbers or words, but they can’t contain spaces or tabs.
```
Comment (bundfussr) [inline vignettes/writing_vignettes.Rmd:186]: Should backticks be used for `[^1]`?

It seems that it is rendered differently in admiraldev
image

 and admiral
image

Comment (zdz2101) [reply vignettes/writing_vignettes.Rmd:186]: image

strange... it is showing the same as admiral for me

File: vignettes/programming_strategy.Rmd (hunk 2 above)
Comment (zdz2101) [reply vignettes/programming_strategy.Rmd:742]: Perhaps adding a small snippet in this section right [here](https://pharmaverse.github.io/admiral/cran-release/articles/admiral.html#arguments) ?

File: vignettes/programming_strategy.Rmd (hunk 1 above)
Comment (zdz2101) [reply vignettes/programming_strategy.Rmd]: added

File: vignettes/programming_strategy.Rmd (hunk 3 above)
Comment (zdz2101) [reply vignettes/programming_strategy.Rmd]: added

File: vignettes/programming_strategy.Rmd (hunk 4 above)
Comment (zdz2101) [reply vignettes/programming_strategy.Rmd]: added

File: vignettes/programming_strategy.Rmd (hunk 5 above)
Comment (zdz2101) [reply vignettes/programming_strategy.Rmd:756]: added

File: vignettes/writing_vignettes.Rmd (hunk 6 above)
Comment (bundfussr) [reply vignettes/writing_vignettes.Rmd]: Sorry, the issue was on my side. Some of my packages were out of date. After updating them admiraldev was rendered as admiral.

**Review by manciniedoardo — APPROVED** [review]
Great work!

File: R/assertions.R [hunk 7]
```diff
@@ -910,26 +879,14 @@ assert_named_exprs <- function(arg, optional = FALSE) {
...
-  )
+  deprecate_stop("0.5.0", "assert_has_variables()", "assert_data_frame()")
 }
 
 #' Is Argument a Function?
 #'
 #' Checks if the argument is a function and if all expected arguments are
 #' provided by the function.
 #'
-#' @description
```
Comment (zdz2101) [inline R/assertions.R:928]: The roxgyen deprecation stuff was in the wrong place

File: R/assertions.R [hunk 8]
```diff
@@ -1001,20 +959,24 @@ assert_function <- function(arg, params = NULL, optional = FALSE) {
...
 #'
 #' @return
 #' The function throws an error if any elements of `params` is not an argument of
 #' the function given by `arg`
 #'
 #' @export
 assert_function_param <- function(arg, params) {
-  deprecate_warn("0.5.0", "assert_function_param()", "assert_function()", always = TRUE)
-  assert_function(arg = arg, params = params)
+  deprecate_stop("0.5.0", "assert_function_param()", "assert_function()", always = TRUE)
```
Comment (zdz2101) [inline R/assertions.R:979]: Correct function was deprecated, but the roxygen documentation portion of it was misplaced

**Comment by zdz2101** [comment]
@bundfussr @manciniedoardo @bms63 added post-release cleanup stuff while I was at it

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -3,16 +3,17 @@ Thank you for your Pull Request! We have developed this task checklist from the
...
 - [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
 - [ ] Address any updates needed for vignettes and/or templates
-- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
-- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
+- [ ] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
+- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page.
```
[the remark this answers is not in the document]
Comment (zdz2101) [reply .github/pull_request_template.md:14]: bad copy/paste job

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -3,16 +3,17 @@ Thank you for your Pull Request! We have developed this task checklist from the
 Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.
 
 - [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
+- [ ] Increment the development version number by calling `usethis::use_version("dev")`. Ensure on the Pull Request in the `DESCRIPTION` file the development version has moved up by 1, e.g. from `x.x.x.9002` -> `x.x.x.9003`.
```
[the remark this answers is not in the document]
Comment (zdz2101) [reply .github/pull_request_template.md:6]: Just going to undo this based on feedback from other PR in admiral
